terps/hugo/heglk/hemedia.c

```diff
@@ -51,6 +51,60 @@ void hugo_setgametitle(char *t)
 static schanid_t mchannel = NULL;
 static schanid_t schannel = NULL;
 
+#ifdef GARGLK
+struct GlkResource {
+  long offset;
+  int id;
+};
+
+static struct GlkResource glk_resources[8192];
```

Comment:
Why such a large fixed-length array?

Reply:
> Why such a large fixed-length array?

Because I hate dynamic array management in C but didn't want to have my refusal to implement one be the source of media-loading failure... but you're right, of course, that there won't be a game out there even _approaching_ this many resources.  Unless I convert this over to C++ I really don't want to do anything but a fixed-size array; what do you think would be a reasonable "max resources" value?  The old Hugo code had 1024, and even that's probably much larger than it needs to be.

Reply:
A few idea I can think of:

1. Use realloc to resize when needed
2. Implement an actual map in C++, with some helper functions, so that the C terps don't need to handle the map themselves
3. Use the memory pointer as the ID, calling a new Glk function that takes a pointer instead of an image ID.

I think I might like 3 the most. But it would mean keeping the resources files in memory permanently, while it looks like Scare reads the file each time you reference a new image. So that's a change I'd want to make anyway.

On the other hand terps that can load multiple resource files would then need a map for filenames, so they'd still need something... But at least such a map probably wouldn't need to be longer than a dozen entries. (Or are there games with even more resource files than that??)